python/lsst/cell_coadds/_fits.py

```diff
@@ -0,0 +1,407 @@
+# This file is part of cell_coadds.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = (
+    "CellCoaddFitsFormatter",
+    "CellCoaddFitsReader",
+    "writeExplodedCoaddAsFits",
+    "writeMultipleCellCoaddAsFits",
+    "writeStitchedCoaddAsFits",
+)
+
+import os
+from typing import TYPE_CHECKING, Any, Mapping
+
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+import numpy as np
+from astropy.io import fits  # type: ignore[import]
+from lsst.afw.image import ImageD, ImageF
+from lsst.daf.base import PropertySet
+from lsst.geom import Box2I, Extent2I, Point2I
+from lsst.obs.base.formatters.fitsGeneric import FitsGenericFormatter
+from lsst.skymap import Index2D
+
+from ._common_components import CoaddUnits, CommonComponents
+from ._identifiers import CellIdentifiers, PatchIdentifiers
+from ._image_planes import OwnedImagePlanes
+from ._multiple_cell_coadd import MultipleCellCoadd, SingleCellCoadd
+from ._uniform_grid import UniformGrid
+
+if TYPE_CHECKING:
+    from ._exploded_coadd import ExplodedCoadd  # pragma: no cover
+    from ._stitched_coadd import StitchedCoadd  # pragma: no cover
+
+
+class CellCoaddFitsFormatter(FitsGenericFormatter):
+    """Interface for writing and reading cell coadds to/from FITS files.
+
+    This assumes the existence of readFits and writeFits methods (for now).
+    """
+
+
+class CellCoaddFitsReader:
+    """A reader class to read from a FITS file and produce cell-based coadds.
+
+    This reader class has read methods that can either return a single
+    component without reading the entire file (e.g., readBBox, readWcs)
+    and read methods that return a full coadd (e.g.,
+    readAsMultipleCellCoadd, readAsExplodedCellCoadd, readAsStitchedCoadd).
+
+    Parameters
+    ----------
+    filename : `str`
+        The name of the FITS file to read.
+    """
+
+    def __init__(self, filename: str) -> None:
+        if not os.path.exists(filename):
+            raise FileNotFoundError(f"File {filename} not found")
```

Comment:
The behavior in `fitsGeneric` seems to be that is silently returns None if the file does not exist.
https://github.com/lsst/obs_base/blob/199db6b3febad54ded76da84192a0a8730815cef/python/lsst/obs/base/formatters/fitsGeneric.py#L54

Should we aim for the same behavior here?

Reply:
It looks like that behavior is only about supporting reading assembling composite datasets from per-file components (which we don't care about for this type, since it'll never be a component), so we don't need to replicate it here.